File: iosApp/iosApp/presentationLayer/pages/familyList/viewModels/FamilyListViewModelMocked.swift
import Foundation
import shared

class FamilyListViewModelMocked: FamilyListViewModelProtocol {
    var viewState: FamilyListViewState = .success

    var familyListModels: [FamilyListModel] = Samples.FamilyList.companion.list1
    var familyListModelsGrouped: [FamilyListModelsGrouped] = {
        Dictionary(grouping: Samples.FamilyList.companion.list1) { item in
            item.isCompletedDate?.toDateNoTime() ?? GlobalStateKt.defaultLocalDateTime.toDateNoTime()
        }.map { key, items in
            FamilyListModelsGrouped(id: key, items: items)
        }.sorted { $0.id > $1.id }
    }()

    var isShowingBarcodeBottomSheet: Bool = false
    var isShowingOpenImageBottomSheet: Bool = false

    var newItemName: String = ""
    var tabIndex: FamilyListPageTabEnum = .pending
    var selectedItemUuid: String = ""
    var openImageSelectedItem: FamilyListModel? = nil

    var sumOfPrioritized: Double = 1.11
    var sumOfPending: Double = 2.22
    var sumOfCompleted: Double = 3.33

    var goToSetting: () -> Void = {}
    var goToQuickInsert: () -> Void = {}
    var goToBarcodeScanner: () -> Void = {}

    func loadData(fromNetwork: Bool, showLoading: Bool) {}
    func add() {}
    func addBy(barcode: String) {}
    func showError(e: Error) {}
    func remove(uuid: String) {}
    func updateIsCompleted(uuid: String, isCompleted: Bool) {}
    func updateIsPrioritized(uuid: String, isPrioritized: Bool) {}
    func updateName(uuid: String, name: String) {}
    func updateQuantity(uuid: String, quantity: Int) {}
    func updatePrice(uuid: String, price: Double) {}
    func openImage(item: FamilyListModel) {}
}

File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/presentationLayer/pages/familyList/viewmodels/FamilyListViewModelMocked.kt
package dev.haroldjose.familysharedlist.android.presentationLayer.pages.familyList.viewmodels

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.lifecycle.ViewModel
import dev.haroldjose.familysharedlist.android.presentationLayer.pages.familyList.views.FamilyListPageTabEnum
import dev.haroldjose.familysharedlist.defaultLocalDateTime
import dev.haroldjose.familysharedlist.domainLayer.extensions.Samples
import dev.haroldjose.familysharedlist.domainLayer.models.FamilyListModel
import kotlinx.datetime.LocalDate

class FamilyListViewModelMocked: ViewModel(), IFamilyListViewModel {
    override val familyListModelsGrouped: Map<LocalDate, List<FamilyListModel>> by mutableStateOf(
        Samples.FamilyList.list1.groupBy { it.isCompletedDate?.date ?: defaultLocalDateTime.date }
    )
    override var familyListModels: List<FamilyListModel> = Samples.FamilyList.list1
    override var loading: Boolean = false
    override var newItemName: String = "newItem"
    override var selectedItemUuid: String = ""
    override var quantity: Int = 1
    override var tabIndex: FamilyListPageTabEnum = FamilyListPageTabEnum.PRIORIZED
    override var openImageSelectedItem: FamilyListModel? = Samples.FamilyList.nutella

    override var goToSetting: () -> Unit = {}
    override var goToQuickInsert: () -> Unit = {}

    override var sumOfPrioritized: Double = 0.0
    override var sumOfPending: Double = 1.23
    override var sumOfCompleted: Double = 44.92

    override suspend fun loadData(fromNetwork: Boolean) {}
    override suspend fun add() {}
    override suspend fun addBy(barcode: String) {}

    override fun showError(e: Throwable) {}
    override suspend fun remove(uuid: String) {}

    override suspend fun updateIsCompleted(uuid: String, isCompleted: Boolean) {}

    override suspend fun updateIsPrioritized(uuid: String, isPrioritized: Boolean) {}

    override suspend fun updateName(uuid: String, name: String) {}

    override suspend fun updateQuantity(uuid: String, quantity: Int) {}

    override suspend fun updatePrice(uuid: String, price: Double) {}

    override fun openImage(item: FamilyListModel) {}
}